ffi/diplomat/src/properties_names.rs

```diff
@@ -0,0 +1,190 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::provider::ffi::ICU4XDataProvider;
+    use alloc::boxed::Box;
+    use icu_properties::{names::PropertyValueNameToEnumMapper, provider, GeneralCategoryGroup};
+    use icu_provider::prelude::{DataProvider, DataResponse, KeyedDataMarker};
+
+    use crate::errors::ffi::ICU4XError;
+
+    /// A type capable of looking up a property value from a string name.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::properties::names::PropertyValueNameToEnumMapper, Struct)]
+    #[diplomat::rust_link(icu::properties::names::PropertyValueNameToEnumMapperBorrowed, Struct)]
+    #[diplomat::rust_link(
+        icu::properties::names::PropertyValueNameToEnumMapper::as_borrowed,
+        FnInStruct,
+        hidden
+    )]
+    #[diplomat::rust_link(
+        icu::properties::names::PropertyValueNameToEnumMapper::from_data,
+        FnInStruct,
+        hidden
+    )]
+    pub struct ICU4XPropertyValueNameToEnumMapper(PropertyValueNameToEnumMapper<u16>);
+
+    fn load<M: KeyedDataMarker<Yokeable = provider::PropertyValueNameToEnumMapV1<'static>>>(
+        provider: &ICU4XDataProvider,
+    ) -> Result<Box<ICU4XPropertyValueNameToEnumMapper>, ICU4XError> {
+        let data = DataProvider::<M>::load(&provider.0, Default::default())
+            .and_then(DataResponse::take_payload)?;
+        Ok(Box::new(ICU4XPropertyValueNameToEnumMapper(
+            PropertyValueNameToEnumMapper::from_data(data),
+        )))
+    }
+
+    impl ICU4XPropertyValueNameToEnumMapper {
+        /// Get the property value matching the given name, using strict matching
+        ///
+        /// Returns -1 if the name is unknown for this property
+        // #[diplomat::rust_link(icu::properties::names::PropertyValueNameToEnumMapperBorrowed::get_strict, FnInStruct)]
+        // #[diplomat::rust_link(icu::properties::names::PropertyValueNameToEnumMapperBorrowed::get_strict_u16, FnInStruct, hidden)]
```

Comment:
Question: Why are these rust links commented out? (here and below)

Reply:
oh, must have happened by mistake

Reply:
https://github.com/unicode-org/icu4x/pull/3202